packages/next/next-server/server/api-utils.ts

```diff
@@ -235,7 +235,9 @@ export function redirect(
     url = statusOrUrl
     statusOrUrl = 307
   }
-
+  if (typeof statusOrUrl === 'number' && typeof url === 'undefined') {
```

Comment:
Perhaps you can do
```suggestion
  if (typeof statusOrUrl === 'number' && typeof url !== 'string') {
```
to capture more invalid cases?

Reply:
I agree. Your solution covers a larger number of invalid cases : )